src/lib/coda_net2/coda_net2.ml

```diff
@@ -57,6 +57,47 @@ type stream_state =
 
 type erased_magic = [`Be_very_careful_to_be_type_safe]
 
+module Go_log = struct
+  let ours_of_go lvl =
+    let open Logger.Level in
+    match lvl with
+    | 0 (* Critical gets mapped to error *) | 1 ->
+        Error
+    | 2 ->
+        Warn
+    | 3 (* Notice gets mapped to info (I can't find any use of notice?) *) | 4
+      ->
+        Info
+    | 5 ->
+        Debug
+    | _ ->
+        Spam
+
+  (* there should be no other levels. *)
+
+  type record =
+    { id: int64
+    ; time: string
+    ; module_: string [@key "module"]
+    ; level: int
+    ; message: string }
+  [@@deriving of_yojson]
+
+  let record_to_message r =
+    Logger.Message.
+      { timestamp= Time.of_string r.time
+      ; level= ours_of_go r.level
+      ; source=
+          Some
+            (Logger.Source.create
+               ~module_:(sprintf "Libp2p_helper.Go.%s" r.module_)
+               ~location:"(not tracked)")
+      ; message= r.message
+      ; metadata=
+          String.Map.of_alist_exn
```

Comment:
Nit: `String.Map.singleton` would be a little more readable.

Reply:
Ah, that's a nice one!